api/proto/teleport/userloginstate/v1/userloginstate.proto

```diff
@@ -40,4 +40,7 @@ message Spec {
 
   // user_type is the type of user this state represents.
   string user_type = 3;
+
+  // original_roles are the user roles that the user originally had prior to the user login state being generated.
```

Comment:
When is a user login state generated? Is this prior to getting access from an access list?

Reply:
I agree with Zac. Some more details here would be helpful